src/amo/pages/UserProfile/index.js

```diff
@@ -101,7 +119,12 @@ export class UserProfileBase extends React.Component<InternalProps> {
       dispatch(
         fetchUserAccount({
           errorHandlerId: errorHandler.id,
-          userId: Number(params.userId),
+          // We should use `Number()` here but we need to fetch users by
+          // username in order to send a server redirect (to support previous
+          // URLs with usernames). That is why we have to ignore the Flow error
+          // here.
+          // $FLOW_IGNORE
+          userId: params.userId,
```

Comment:
This adds the ability to fetch a user based on a string username, in addition to a numeric userid. Do you think it's worth adding a test with a string username to the suite for this? I notice that there isn't one in there.

Reply:
There is a test case for that, no? https://github.com/mozilla/addons-frontend/blob/f3d02de14effb6a4d0c100fc9ae9afa3ab57c1c4/tests/unit/amo/pages/TestUserProfile.js#L1034

Reply:
Nevermind, I got it: you meant a test case to fetch a user, before the redirect.